src/include/txn_inline.h

```diff
@@ -145,11 +145,11 @@ __txn_resolve_prepared_update(WT_SESSION_IMPL *session, WT_UPDATE *upd)
      *
      * As updating timestamp might not be an atomic operation, we will manage using state.
      */
-    upd->prepare_state = WT_PREPARE_LOCKED;
-    WT_WRITE_BARRIER();
+    atomic_store_explicit(&upd->prepare_state, WT_PREPARE_LOCKED, memory_order_relaxed);
```

Comment:
I thought the store associated with a release barrier should follow the barrier, and the reverse for read/acquire barriers.

Reply:
I looked at the doc again. I think you are right. But in this particular example, we want to ensure the lock of the prepared state happens before the writes to the timestamps. If we put the fence before the lock of the prepared state, it's not helping that. May be instead of using the relaxed write here:

```
atomic_store_explicit(&upd->prepare_state, WT_PREPARE_LOCKED, memory_order_relaxed);
```
 
We should use a release write as well then followed by a release fence.

Reply:
Changed the relaxed store to use the release store.